enhancements/ingress/admin-configured-ca-bundle.md

```diff
@@ -0,0 +1,243 @@
+---
+title: admin-configured-ca-bundle
+authors:
+  - "@bhb"
+reviewers:
+  - "@Miciah" ## reviewer for cluster-ingress-operator component
+  - "@tgeer" ## reviewer for CFE
+approvers:
+  - "@jerpeter1" ## approver for CFE
+  - "@Miciah" ## approver for cluster-ingress-operator component
+api-approvers:
+  - None
+creation-date: 2023-11-02
+last-updated: 2023-11-02
+tracking-link:
+  - https://issues.redhat.com/browse/RFE-2182
+  - https://issues.redhat.com/browse/OCPSTRAT-431
+  - https://issues.redhat.com/browse/NE-761
+see-also:
+  - None
+replaces:
+  - None
+superseded-by:
+  - None
+---
+
+# Support for admin configured CA trust bundle in Ingress Operator
+
+## Summary
+
+This enhancement describes the proposal to allow an administrator of OpenShift
+to configure trusted Certificate Authorities (CA) to verify the service's 
+certificate by OpenShift Router for the re-encrypted traffic.
+
+## Motivation
+
+When an OpenShift administrator creates a route with the re-encrypt termination
+type, they have to also provide a CA certificate for that route if it uses a
+custom CA instead of the default OpenShift CA for certificate validation.
+In addition, when the same CA is used by the administrator for signing several
+services' certificates it is time-consuming and error-prone to configure the
+same CA certificate in every created route object. Likewise, we'd like to improve
+the process for updating a CA certificate when expired, compromised or for any other
+reason in each route object. Instead, this document proposes a simpler alternative:
+a feature in which an administrator could define a CA bundle for verifying certificates
+from a ConfigMap consisting of the list of CA certificates, associated OpenShift Ingress
+Controller, and trusted OpenShift CA.
+
+### User Stories
+- As an OpenShift cluster administrator, I want to configure admin or the enterprise
+  CA certificate bundle which should be used for verifying service's certificate by 
+  the OpenShift router, instead of configuring the same as destination CA certificate 
+  in each of the created route object.
+- As an OpenShift developer, I want to create a CA trust bundle for Ingress operator usage,
+  which includes the OpenShift service CA and the admin provided CA bundle as the default
+  CA for verifying the service's certificate when a destination CA certificate is not
+  configured for a route.
+- As an OpenShift developer, I want to update the OpenShift router certificate cache whenever
+  the OpenShift service CA certificate or the admin configured CA certificate bundle is 
+  updated or deleted.
+
+### Goals
+
+- An OpenShift administrator can configure a CA certificate bundle and expect
+  that to be used for verifying the service's certificate by the OpenShift
+  router without defining the destination CA certificate in the route object
+  created for the service.
+- Openshift Router should be updated seamlessly when the OpenShift service
```

Comment:
Please define "seamlessly".  When the certificate data changes, the container needs to restart to pick up the new data, so it is not absolutely seamless.

Reply:
Since router updates the new certificate data automatically upon detecting changes(included as part of the proposal) and no action from user is required apart from updating the certificate bundle and reload option provided by the router for updating the certificate data doesn't require container to be restarted, I thought to mention seamlessly. Please let me know what you think.

Reply:
I believe the router reloads (my mistake saying that the container restarts) when the backing files change, even if they are volume-mounted, but I could be wrong.  Have you observed otherwise?

Reply:
No, I have the same understanding. A watcher is added for the volume-mounted ca-bundle file in the Openshift Router and on detecting the changes it reloads the config.